akka-http/src/main/scala/akka/http/scaladsl/server/directives/FormFieldDirectives.scala

```diff
@@ -40,6 +59,50 @@ trait FormFieldDirectives extends ToNameReceptacleEnhancements {
 }
 
 object FormFieldDirectives extends FormFieldDirectives {
+
+  val _formFieldSeq: Directive1[immutable.Seq[(String, String)]] = {
+    import BasicDirectives._
+    import FutureDirectives._
+    import akka.http.scaladsl.unmarshalling._
+
+    extract { ctx ⇒
+      import ctx.{ executionContext, materializer }
+      Unmarshal(ctx.request.entity).to[StrictForm].fast.flatMap { form ⇒
+        val fields = form.fields.collect {
+          case (name, field) if name.nonEmpty ⇒
+            Unmarshal(field).to[String].map(fieldString ⇒ (name, fieldString))
```

Comment:
That's a valid point that we should probably address generally, not just here.
Should we really allow the `PredefinedFromEntityUnmarshallers` to buffer incoming data without any configured upper bound (which is what is currently happening)?
I guess it makes sense to introduce a new kind of config setting that we make available implicitly and that the `PredefinedFromEntityUnmarshallers` depend on to bail out if accumulation goes beyond the configured threshold.

Once we have this the implementation that @wjur proposes here becomes less frightening...
(Thanks, @wjur, for taking this on and provoking these interesting discussions!)